main.go

```diff
@@ -25,6 +25,15 @@ import (
 )
 
 func main() {
+	// NOTE(jalextowle): Since `os.Exit` will cause the process to exit, this defer
+	// must be at the bottom of the defer stack to allow all other defer calls to
+	// be called first. We pass this deferred function a pointer to an exit code
+	// so that it can be altered later in the program.
+	exitCode := 0
+	defer func(code *int) {
+		os.Exit(*code)
+	}(&exitCode)
```

Comment:
I don't believe a pointer is necessary here. We can just do

```go
defer func() {
  if code != 0 {
    os.Exit(code)
  }
}()
```

https://play.golang.org/p/Pcn9XIJ2vjH

Also, let's only explicitly run os.Exit() only if an error happens. And let the default be to do nothing.

Reply:
Good call 👍